.github/workflows/test-build-deploy.yml

```diff
@@ -342,27 +346,39 @@ jobs:
         run: |
           docker load -i ./mimir_race_image
           export IMAGE_TAG_RACE=$(make image-tag-race)
-          docker run "grafana/mimir:$IMAGE_TAG_RACE" --version
+          docker run "grafana/mimir:distroless-$IMAGE_TAG_RACE" --version
+          if ${{ needs.checkDeploy.outputs.is_deploy == 'true' }}; then
+            docker load -i ./mimir_race_image_alpine
+            docker run "grafana/mimir:$IMAGE_TAG_RACE"  --version
+          fi
       - name: Preload Images
         # We download docker images used by integration tests so that all images are available
         # locally and the download time doesn't account in the test execution time, which is subject
         # to a timeout
         run: go run ./tools/pre-pull-images | xargs -n1 -P4 docker pull
       - name: Integration Tests
         run: |
-          export IMAGE_TAG_RACE=$(make image-tag-race)
-          export MIMIR_IMAGE="grafana/mimir:$IMAGE_TAG_RACE"
-          export IMAGE_TAG=$(make image-tag)
-          export MIMIRTOOL_IMAGE="grafana/mimirtool:$IMAGE_TAG"
-          export MIMIR_CHECKOUT_DIR="/go/src/github.com/grafana/mimir"
-          echo "Running integration tests with image: $MIMIR_IMAGE (Mimir), $MIMIRTOOL_IMAGE (Mimirtool)"
-          echo "Running integration tests (group ${{ matrix.test_group_id }} of ${{ matrix.test_group_total }}) with Go version: $(go version)"
-          ./.github/workflows/scripts/run-integration-tests-group.sh --index ${{ matrix.test_group_id }} --total ${{ matrix.test_group_total }}
+          ./.github/workflows/scripts/run-integration-tests-group.sh --index ${{ matrix.test_group_id }} --total ${{ matrix.test_group_total }} --tagprefix "distroless-"
+      - name: Run Integration Tests on alpine when deploying
+        if: ${{ needs.checkDeploy.outputs.is_deploy == 'true' }}
+        run: |
+          ./.github/workflows/scripts/run-integration-tests-group.sh --index ${{ matrix.test_group_id }} --total ${{ matrix.test_group_total }} --tagprefix ""
 
+  checkDeploy: 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Whether In Deploy Process
+        id: check_deploy
+        run: |
+          is_deploy=${{ (startsWith(github.ref, 'refs/tags/') || startsWith(github.ref, 'refs/heads/r')) && github.event_name == 'push' && github.repository == 'grafana/mimir' }}
```

Comment:
I think instead of a "job" here, you could evaluate this as a github expression into an [env](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#env) var. This env-var could then be accessed on a per-job (or even per-step) basis as `if: env.is_deploy`.

Reply:
I was slightly wrong about the env option. The env vars can only be strings so evaluating `is_deploy` still isn't very neat. So I think keeping it as a job is probably best.